pkg/route/apis/route/validation/validation.go

```diff
@@ -22,16 +22,45 @@ var ValidateRouteName = apimachineryvalidation.NameIsDNSSubdomain
 
 // ValidateRoute tests if required fields in the route are set.
 func ValidateRoute(route *routeapi.Route) field.ErrorList {
+	return validateRoute(route, true, false)
+}
+
+func invalidDNS1123LabelCheck(route *routeapi.Route) bool {
+	segments := strings.Split(route.Spec.Host, ".")
+	for _, s := range segments {
+		if len(kvalidation.IsDNS1123Label(s)) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// validateRoute - private function to validate route
+func validateRoute(route *routeapi.Route, hostnameUpdated, skipDNS1123Validation bool) field.ErrorList {
```

Comment:
```suggestion
func validateRoute(route *routeapi.Route, checkHostname bool) field.ErrorList {
```
Only need to check the hostname if it was updated && was valid before the change